addons/mail/models/mail_thread.py

```diff
@@ -4154,24 +4154,34 @@ def _get_mail_redirect_suggested_company(self):
             return self.company_id
         return False
 
+    def _get_mail_thread_access_rights(self):
+        res = {}
+        for thread in self:
+            access_rights = {
+                'hasWriteAccess': False,
+                'hasReadAccess': True,
+                'canPostOnReadonly': thread._mail_post_access == 'read',
+            }
+            try:
+                thread.check_access_rule("write")
+                thread.check_access_rights("write")
+                access_rights['hasWriteAccess'] = True
+            except AccessError:
+                pass
+            res[(thread._name, thread.id)] = access_rights
```

Comment:
Why would we return a dict ? Can't we return a list ordered as self ? Looking at callers it would be easier to invoke.